map_test.go

```diff
@@ -1720,15 +1720,74 @@ func TestMapPinning(t *testing.T) {
 	if value != 42 {
 		t.Fatal("Pinning doesn't use pinned maps")
 	}
+}
+
+func TestMapPinningIncompatible(t *testing.T) {
+	tmp := testutils.TempBPFFS(t)
+	c := qt.New(t)
+
+	m1, err := NewMapWithOptions(&MapSpec{
+		Name:       "test",
+		Type:       Hash,
+		KeySize:    4,
+		ValueSize:  4,
+		MaxEntries: 1,
+		Pinning:    PinByName,
+	}, MapOptions{PinPath: tmp})
+	if err != nil {
+		t.Fatal("Can't create map:", err)
+	}
+	defer m1.Close()
 
-	spec.KeySize = 8
-	m3, err := NewMapWithOptions(spec, MapOptions{PinPath: tmp})
-	if err == nil {
-		m3.Close()
-		t.Fatalf("Opening a pinned map with a mismatching spec did not fail")
+	tests := []struct {
+		name string
+		spec *MapSpec
+		test func(*qt.C, *IncompatibleMapError)
```

Comment:
Why not `testing.T`?
```suggestion
		test func(*testing.T, *IncompatibleMapError)
```

qt.C is kind of exotic and might be removed by the next version of the package.

Reply:
Sure. You'll notice I was using `c.Run()` since there are calls to `c.Assert` in the subtests. Subtests shouldn't call Fatal or Fail on the parent test scope, that leads to panics:

`testing.go:1490: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test`